zerver/lib/actions.py

```diff
@@ -2737,6 +2763,28 @@ def our_person(p):
                         bot.update(event['bot'])
 
         elif event['type'] == 'stream':
+            if event['op'] == 'create':
+                for stream in event['streams']:
+                    stream_name = stream['name']
+                    realm = get_realm(user_profile.realm.domain)
+                    new_stream = get_stream(stream_name, realm)
+                    if not stream['invite_only']:
```

Comment:
Might as well move this check up a few lines, since the preceding variables are not used unless `invite_only === True`